web/src/modules/topic/components/CriteriaTable/CriteriaTable.tsx

```diff
@@ -86,63 +83,63 @@ export const CriteriaTable = ({ problemNodeId }: Props) => {
   const columnNodes = useSolutionsForColumns ? solutions : criteria;
 
   const rowData = buildRows(rowNodes, columnNodes, criterionSolutionEdges);
-  const columnData = buildColumns(columnNodes);
+  const columnData = buildColumns(problemNode, columnNodes);
 
   return (
     <>
-      <PositionedCloseButton onClick={() => closeTable()} color="primary">
-        <Cancel />
-      </PositionedCloseButton>
-
-      <TitleDiv>
-        <Typography variant="h4">Criteria for solving:</Typography>
-        <EditableNode node={problemNode} />
-      </TitleDiv>
-
-      <TableDiv numberOfColumns={columnData.length}>
-        {/* Hard to tell if material-react-table is worth using because the cells are all custom components. */}
-        {/* It's doubtful that we'll use sorting/filtering... but pinning and re-ordering may come in handy. */}
-        <MaterialReactTable
-          columns={columnData}
-          data={rowData}
-          enableColumnActions={false}
-          enablePagination={false}
-          enableBottomToolbar={false}
-          enableTopToolbar={false}
-          enableSorting={false}
-          muiTablePaperProps={{
-            className: "criteria-table-paper",
-          }}
-          initialState={{ columnPinning: { left: ["rowNode.data.label"] } }}
-        />
-
-        <StyledTransposeTableButton
-          size="small"
-          variant="contained"
-          color="neutral"
-          onClick={() => setUseSolutionsForColumns(!useSolutionsForColumns)}
-        >
-          <PivotTableChart />
-        </StyledTransposeTableButton>
-
-        <PositionedAddNodeButton
-          position={useSolutionsForColumns ? "column" : "row"}
-          fromNodeId={problemNodeId}
-          as="child"
-          toNodeType="criterion"
-          relation="criterion for"
-        />
-
-        <PositionedAddNodeButton
-          position={useSolutionsForColumns ? "row" : "column"}
-          fromNodeId={problemNodeId}
-          as="child"
-          toNodeType="solution"
-          relation="solves"
-        />
-      </TableDiv>
-
       <Global styles={tableStyles} />
+      <MaterialReactTable
+        columns={columnData}
+        data={rowData}
+        enableColumnActions={false}
+        enablePagination={false}
+        enableBottomToolbar={false}
+        renderToolbarInternalActions={({ table }) => {
```

Comment:
I think it'll read a little nicer if we extract a `ToolbarActions` component (putting it just below the `columnData` definition seems fine) to pass to this prop, because right now you can't see all the of the table's props without scrolling